### PR TITLE
Adding "Azure_CloudService_SI_Disable_RemoteDesktop_Access" remediati…

### PR DESCRIPTION
**Pull request by RohitYadav-msft**
…on script

### CODE REVIEW

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:13]: Please confirm if just a `Contributor` role would suffice.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:53]: Nit: Please check the indentation.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:60]: Can lines 59 - 90 replaced with a loop?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:218]: Do we change the path to `%temp% or the current directory?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:238]: Do we need to check only "enabled" extensions?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:251]: There can be two cases:
- Extension is not present.
- Extension is present, but, disabled.
Do we consider both of them for the "disabled" use case?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:350]: Nit: Please fix the indentation.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:292]: Please document that the script does not currently remediate Cloud Services that had RDP enabled during the deployment, in which case, no extensions would be present.

**Review by gvaradarajan-msft — CHANGES_REQUESTED** [review]

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: We can execute remediation script using contributor permission, have added changes for this

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:53]: Indentation looks good in VS, 
will resolve this comments at the end from portal

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: Done

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: Done

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: Yes

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:285]: Added

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:349]: Will do at the end from portal

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:244]: Covered this  scenario.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:299]: Are you breaking on first failure?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:304]: Are lines 304-307 valid?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:314]: Do you want to include the extension details here?

**Review by gvaradarajan-msft — CHANGES_REQUESTED** [review]
Please include another reviewer.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:299]: We have used 'For' loop to remove extensions from cloud service(s), so if we failed to remove any extension(due to permission issue or cmd. failure) then we are breaking flow of removing other extension of that cloud service and simply storing service and RG name in an array(containing list of cloud services skipped from remediation).

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:304]: Yes, here we are printing table, containing service and RG name on which we have successfully performed remediation.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:314]: Here we are showing service and RG name of cloud services for which we are unable to do remediation(due to permission issue or any command failure).
Therefore showing extension details is not required

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: Do you want to provide the `UninstallConfiguration` flag to this command?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:131]: Please include a note that says - These need to be remediated via Azure Portal.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (gvaradarajan-msft) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:132]: Is rollback supported? If not yet, and if rollback is not possible, please include a prompt that asks for the user's input before proceeding.

**Review by gvaradarajan-msft — CHANGES_REQUESTED** [review]

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:132]: Added

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: Added

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: Not considering for now, we can take it as enhancement.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:228]: Does this return RDP extension for all roles in a slot?

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:131]: We should output list of such cloud services.

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:228]: Yes

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (RohitYadav-msft) [reply Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1]: Girish will be working on this to output list of service on which RDP enabled while doing deployment

**Review by gvaradarajan-msft — APPROVED** [review]
Please get a second review.

**Review by TarunKrShukla — APPROVED** [review]

File: Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1
Comment (TarunKrShukla) [inline Scripts/RemediationScripts/Remediate-RemoteDesktopAccess.ps1:338]: In this case for some of the cloud services we might have be able to remove RDP extensions we should print those as well.

**Review by TarunKrShukla — CHANGES_REQUESTED** [review]
Please work on comments.

**Review by TarunKrShukla — APPROVED** [review]

**Pull request merged**
